artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -1975,19 +1970,14 @@ public void expire(final MessageReference ref) throws Exception {
 
    @Override
    public void expire(final MessageReference ref, final ServerConsumer consumer) throws Exception {
-      SimpleString messageExpiryAddress = expiryAddressFromMessageAddress(ref);
-      if (messageExpiryAddress == null) {
-         messageExpiryAddress = expiryAddressFromAddressSettings(ref);
-      }
-
-      if (messageExpiryAddress != null) {
+      if (addressSettings.getExpiryAddress() != null) {
```

Comment:
I looked at the commit for that change and it wasn't clear what it was for since there was no test associated with it. You added a test for it later (i.e. `o.a.a.a.t.i.c.d.ExpireWhileLoadBalanceTest`) which I ran after my changes and it succeeded. It looked like a semantic change, but I couldn't verify it with any existing test. It's still not clear to me what the desired behavior would be for messages that expire in a SnF queue. Are they supposed to go to the expiry address configured for where they were originally sent? If so, it's not clear to me why that's better than going to the expiry address configured for the SnF.

Reply:
ok.. thanks.. I just wanted to make sure you looked into that.


I'm running the whole test suite to make sure.